### PR TITLE
infoschema_v2: init schema cache size in meta

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: ref #50959 

Problem Summary:

### What changed and how does it work?
- init tidb_schema_cache_size in meta
- load it from meta before bootstrap
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/domain/domain_sysvars.go [hunk 1]
```diff
@@ -118,8 +120,16 @@ func (do *Domain) getExternalTimestamp(ctx context.Context) (uint64, error) {
 	return do.store.GetOracle().GetExternalTimestamp(ctx)
 }
 
-func (do *Domain) changeSchemaCacheSize(size uint64) {
+func (do *Domain) changeSchemaCacheSize(size uint64) error {
+	err := kv.RunInNewTxn(kv.WithInternalSourceType(context.Background(), kv.InternalTxnDDL), do.store, true, func(_ context.Context, txn kv.Transaction) error {
```
Comment (tangenta) [inline pkg/domain/domain_sysvars.go:124]:
```suggestion
	err := kv.RunInNewTxn(kv.WithInternalSourceType(do.ctx, kv.InternalTxnDDL), do.store, true, func(_ context.Context, txn kv.Transaction) error {
```

Maybe add a field `ctx` to `Domain`, and initialize it in `Domain.Init()`.

Comment (tangenta) [reply pkg/domain/domain_sysvars.go:124]: Or move `Domain.Init()` ctx initialization to `NewDomain`.

Comment (GMHDBJD) [reply pkg/domain/domain_sysvars.go:124]: e67b16a use the context pass by setVar

File: pkg/session/session.go [hunk 2]
```diff
@@ -3398,6 +3424,10 @@ func bootstrapSessionImpl(store kv.Storage, createSessionsImpl func(store kv.Sto
 	if err != nil {
 		return nil, err
 	}
+	err = InitTiDBSchemaCacheSize(store)
+	if err != nil {
+		logutil.BgLogger().Fatal("init tidb schema cache size error", zap.Error(err))
```
Comment (tangenta) [inline pkg/session/session.go:3429]:
```suggestion
		return nil, err
```

The caller should be able to handle this.

**Review by tangenta — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**